Add Cjk width token filter (#7917)
opensearch-project/documentation-website: _analyzers/token-filters/cjk-width.md

---
layout: default
title: CJK width
parent: Token filters
nav_order: 40
---

# CJK width token filter

> The CJK Width token filter in OpenSearch normalizes Chinese, Japanese, and Korean (CJK) tokens by converting full-width ASCII Character to their standard (half-width) ASCII equivalents and half-width Katakana characters to their full-width.

Reviewer: ```suggestion
The CJK Width token filter normalizes Chinese, Japanese, and Korean (CJK) tokens by converting full-width ASCII characters to their standard (half-width) ASCII equivalents and half-width katakana characters to their full-width equivalents.
```

>  - **Converting full-width ASCII Character**: In CJK texts, ASCII characters (such as letters and numbers) can appear in full-width form, which occupies the space of two half-width characters. Full-width ASCII characters are typically used in East Asian typography to align with the width of CJK characters. However, for the purpose of indexing and searching, these full-width characters need to be normalized to their standard (half-width) ASCII equivalents.

Reviewer: Could you make these headings level 3?

>     See following example:

Reviewer: ```suggestion
The following example illustrates ASCII character normalization:
```

>         Full-Width: ＡＢＣＤＥ １２３４５

Reviewer: I think these are better formatted as a code block and aligned so the letter A is approximately at the same column in both lines (lines 16 and 17)

>         Normalized: (Half-Width): ABCDE 12345

Reviewer: ```suggestion
        Normalized (half-width): ABCDE 12345
```

>  - **Converting half-width Katakana characters**: The CJK Width token filter converts half-width Katakana characters to their full-width counterparts, which are the standard form used in Japanese text. This normalization is important for consistency in text processing and searching.

Reviewer: ```suggestion
 - **Converting half-width katakana characters**: The CJK width token filter converts half-width katakana characters to their full-width counterparts, which are the standard form used in Japanese text. This normalization is important for consistency in text processing and searching.
```

>     See following example:

Reviewer: ```suggestion
The following example illustrates ASCII converting half-width katakana characters:
```

>         Half-Width "Katakana": ｶﾀｶﾅ

Reviewer: ```suggestion
        Half-Width katakana: ｶﾀｶﾅ
```

>         Normalized (Full-Width "Katakana"): カタカナ

Reviewer: ```suggestion
        Normalized (full-width) katakana: カタカナ
```

---
layout: default
title: CJK width
parent: Token filters
nav_order: 40
---

# CJK width token filter

The CJK Width token filter normalizes Chinese, Japanese, and Korean (CJK) tokens by converting full-width ASCII characters to their standard (half-width) ASCII equivalents and half-width katakana characters to their full-width equivalents.

### Converting full-width ASCII Character

In CJK texts, ASCII characters (such as letters and numbers) can appear in full-width form, which occupies the space of two half-width characters. Full-width ASCII characters are typically used in East Asian typography to align with the width of CJK characters. However, for the purpose of indexing and searching, these full-width characters need to be normalized to their standard (half-width) ASCII equivalents.

> The following example illustrates ASCII character normalization:

Reviewer: ```suggestion
The following example illustrates ASCII character normalization:

```

```
        Full-Width:              ＡＢＣＤＥ １２３４５
        Normalized (half-width): ABCDE 12345

> ```

Reviewer: ````suggestion
```

````

### Converting half-width katakana characters

The CJK width token filter converts half-width katakana characters to their full-width counterparts, which are the standard form used in Japanese text. This normalization is important for consistency in text processing and searching.

> The following example illustrates ASCII converting half-width katakana characters:

Reviewer: ```suggestion
The following example illustrates ASCII converting half-width katakana characters:

```

---
layout: default
title: CJK width
parent: Token filters
nav_order: 40
---

# CJK width token filter

> The CJK Width token filter normalizes Chinese, Japanese, and Korean (CJK) tokens by converting full-width ASCII characters to their standard (half-width) ASCII equivalents and half-width katakana characters to their full-width equivalents.

Reviewer 2: ```suggestion
The `cjk_width` token filter normalizes Chinese, Japanese, and Korean (CJK) tokens by converting full-width ASCII characters to their standard (half-width) ASCII equivalents and half-width katakana characters to their full-width equivalents.
```

> ### Converting full-width ASCII Character

Reviewer 2: ```suggestion
### Converting full-width ASCII characters
```

> In CJK texts, ASCII characters (such as letters and numbers) can appear in full-width form, which occupies the space of two half-width characters. Full-width ASCII characters are typically used in East Asian typography to align with the width of CJK characters. However, for the purpose of indexing and searching, these full-width characters need to be normalized to their standard (half-width) ASCII equivalents.

Reviewer 2: ```suggestion
In CJK texts, ASCII characters (such as letters and numbers) can appear in full-width form, occupying the space of two half-width characters. Full-width ASCII characters are typically used in East Asian typography for alignment with the width of CJK characters. However, for the purposes of indexing and searching, these full-width characters need to be normalized to their standard (half-width) ASCII equivalents.
```

The following example illustrates ASCII character normalization:

```
        Full-Width:              ＡＢＣＤＥ １２３４５
        Normalized (half-width): ABCDE 12345
```

### Converting half-width katakana characters

> The CJK width token filter converts half-width katakana characters to their full-width counterparts, which are the standard form used in Japanese text. This normalization is important for consistency in text processing and searching.

Reviewer 2: ```suggestion
The `cjk_width` token filter converts half-width katakana characters to their full-width counterparts, which are the standard form used in Japanese text. This normalization, illustrated in the following example, is important for consistency in text processing and searching:
```

> The following example illustrates ASCII converting half-width katakana characters:

Reviewer 2: ```suggestion
```
